### ISSUE TITLE
Artifacting on Navi10

### ISSUE THREAD

**Issue by mcoffin** [GPUOpen-Drivers/AMDVLK#105]
I've been experiencing extreme artifacting and garbled output when using `AMDVLK-2018.Q3.3` with a Navi10 5700 XT.

The artifacting happens even with the most simple of demo triangle applications, but, (with LLVM 9 and latest mesa git) RADV works fine.

# System Specs

* Processor: AMD Ryzen 9 3900X
* Motherboard: ASUS ROG STRIX x570-F gaming
* RAM: G.Skill Ripjaws 3200
* Kernel: [`amd-staging-drm-next`](https://aur.archlinux.org/packages/linux-amd-staging-drm-next).
* `libdrm`: Latest git as of 10am 20190805 Mountain Time.

# Other things tried

* 5.3-rc3 kernel
* latest `dev` versions of `amdvlk`
* Stable libdrm (segfaults)
* `drm-fixes-5.3` kernel
* `AMDVLK-2019.Q3.2` update (first to support Navi10).
* super aggressive fan curve (100%), via [this kernel patch](https://patchwork.freedesktop.org/patch/320885/) and sysfs/hwmon
* Binary-distributed version of `AMDVLK-2019.Q3.{2,3}`

It seems that the documentation indicates that support is getting there, and other people are reporting better results than I'm getting, so I"m thinking either I must be doing something wrong, or there's still something pretty busted, as even simple triangle demos won't work.

Thanks in advance for any help you can offer.

**Comment by mcoffin** [comment]
As a starting place, is the expected (i.e. y'all know it's not ready yet), or am I experiencing an unexpected issue?

**Comment by VadikLeshy** [comment]
What do the artifacts look like?
Can [this](https://github.com/GPUOpen-Drivers/AMDVLK/issues/106) be related?

**Comment by mcoffin** [comment]
@VadikLeshy They look slightly different from that. In fact, they're almost identical to what I was seeing with mesa 19.2 git RADV up until a few weeks back.

It could be similar though, I'll try to get a screenshot when I get home tonight.

**Comment by MadByteDE** [comment]
I can confirm this behaviour for all games I tested using DXVK(1.3.2). [Here are some screenshots](https://imgur.com/a/3DOWATz).
I don't have any artifacting when using the vulkan-radeon-git driver. 

[Hardware Infos](https://docs.google.com/document/d/1GUX-GAKn4-kLSwy8PBghxg7eUox_D6FuNSXkaVK7udg/edit?usp=sharing)

## I'm using 
- navi10 firmware (26.08.2019)
- latest amd-staging-drm-next
- mesa-git
- latest amdvlk-git
- latest llvm-git
- latest xf86-video-amdgpu-git

**Comment by aqxa1** [comment]
I have the same problem with AMDLVK. I'll also add that extracting the AMDGPU-PRO Vulkan driver (19.30-855429) and running it on top of the Mesa stack (I haven't tested the full pro stack), results in the same bug. So there may be something in common there.

**Comment by VadikLeshy** [comment]
On my APU Ryzen 5 2400G (Raven Ridge) I have a problem very similar (judging by the screenshots) to the @MadByteDE 's one.

https://github.com/GPUOpen-Drivers/AMDVLK/issues/106

**Comment by mcoffin** [comment]
@aqxa1 mesa has fixed this if you pull from the latest git master.

I still experience the issue with AMDVLK. It's a real bummer to be honest. I have yet to find a single person for whom the AMDVLK driver is currently working on their navi10 card

**Comment by mcoffin** [comment]
> [Here are some screenshots](https://imgur.com/a/3DOWATz).

Yep. This is the exact problem that I'm having as well @MadByteDE

**Comment by mcoffin** [comment]
It's also worth noting that `2019.Q3.5` works for me for my RX 590, but not the 5700XT.

**Comment by mcoffin** [comment]
Just tried again with `linux-5.3rc7`, and even on that `amdgpu` driver I'm still getting the same issue(s).

**Comment by mcoffin** [comment]
@MadByteDE @UnicornsOnLSD Do either of you have the 5700 non-XT? Wondering if this is isolated to the 5700 XT or gfx10_0 as a whole

**Comment by UnicornsOnLSD** [comment]
@mcoffin I have the 5700 XT

**Comment by mcoffin** [comment]
I hope at some point we can get a confirmation from one of the AMD devs that they can at least duplicate this problem

**Comment by mcoffin** [comment]
After a lot of work patching `llpc` to use upstream LLVM 9 (with only the needed intrinsics patches from GPUOpen-Drivers/llvm), I can now confirm that all my work was for naught because the issue still persists.

This leads me to believe that one of the following three options is the case.

1. the AMD devs aren't testing against the open source `amdgpu` driver, and there's an interaction issue there.
2. There's a bug in the gfx10-specific code
3. There is something wrong with NGG for AMDVLK, because it doesn't seem to obey the settings option to disable it, so there may be something funky going on there.

**Comment by UnicornsOnLSD** [comment]
@mcoffin Thanks for having a look, I hope AMD looks into this soon and confirms that it is a bug.

**Comment by mcoffin** [comment]
Honestly, just hearing back from the AMD devs saying that they can replicate the problem would go a LONG way at this point. I've spent literally every waking second of the last two days trying to fix it, and while I learned a *ton* about GPU driver design, and the whole AMDVLK ecosystem in general, I still wasn't able to get it working.

I did however manage to update the LLVM version in AMDVLK to be rebased off of the latest LLVM, which is kinda nice.

**Comment by mcoffin** [comment]
Also, if anyone here knows what this kind of corruption even LOOKS like, I'd appreciate any info you can guess at since I'm trying to fix this right now.

**Comment by UnicornsOnLSD** [comment]
Here are the images from my issue, GTA V's artifacting is fullscreen, I just had the graphics settings done incorrectly.
![image](https://user-images.githubusercontent.com/44349936/64450045-2fcdae00-d0d9-11e9-9c6b-085d4fbcb899.png)
![image](https://cdn.discordapp.com/attachments/258997449232744452/619585914963755009/unknown.png)

**Comment by UnicornsOnLSD** [comment]
Judging by [this website](http://www.playtool.com/pages/artifacts/artifacts.html), it kind of looks like VRAM artifacting but that's a pretty uneducated guess. Good luck on your endeavours, @mcoffin :)

**Comment by mcoffin** [comment]
Honestly looks a bit like

**Comment by jinjianrong** [comment]
Thanks all for reporting the issue, We are checking it.

**Comment by qiaojbao** [comment]
I checked the upstream drivers for Navi10, it seems works fine on my ubuntu 18.04.2 system.
I test the vulkaninfo, vkcube, and steam games, like F1 2017, Mad Max, and The Talos Principle.
I will try the DXVK later.

**System:**
Processor:	Intel(R) Core(TM) i7-4790K CPU @ 4.00GHz
Motherboard:	Z77 Extreme4
RAM:		G.Skill DDR3-1600 12GB
Kernel:		v5.3-rc8
AMDVLK:	amdvlk_2019.Q3.5
libdrm:		libdrm-amdgpu1 2.4.99+git1909070630.10cd9c~oibaf~b
libllvm:	        libllvm9 1:9~+rc2-1~exp1~oibaf~b

**Steps:**
1 install ubuntu18.04.2
  apt-get update
  apt-get dist-upgrade
2 Copy Navi10 firmware into /lib/firmware/amdgpu, you can get firmware from
https://people.freedesktop.org/~agd5f/radeon_ucode/navi10/
  Install kernel, you can get kernel v5.3rc8 at 
https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.3-rc8/
  install AMDVLK driver, get it at 
https://github.com/GPUOpen-Drivers/AMDVLK/releases/download/v-2019.Q3.5/amdvlk_2019.Q3.5_amd64.deb
  Reboot system and choose 5.3rc8 kernel.
3 Add PPA and update the latest libllvm, libdrm and so on.
add-apt-repository ppa:oibaf/graphics-drivers
  apt-get update
  apt-get dist-upgrade (will install libllvm9, update libdrm-amdgpu...)
  Reboot system.

**Comment by mcoffin** [comment]
I am also running those running nearly those identical specs @qiaojbao. What's different with that libdrm version (`libdrm-amdgpu1`), compared to just `libdrm-2.4.99`?

That's the only difference I see in our setups at a first glance. I'll be around tonight to do some more testing (possibly with the binary packages you mentioned just to do a complete test).

Lastly, the libllvm version shouldn't matter, given that (by default, without overriding `XGL_LLVM_SRC_PATH`) `AMDVLK` should be using it's own LLVM version, yes? Some of `llpc` doesn't even compile with the upstream LLVM due to the validator not liking the pre-inlined functions.

**Comment by jinjianrong** [comment]
libllvm is only used by MesaGL, not by AMDVLK. 
AMDVLK uses its own LLVM version in amd-vulkan-dev or amd-vulkan-master branch

**Comment by qiaojbao** [comment]
Thanks all for this issue.
